packages/app/lib/ui/widgets/used_for_dedup_tag_list.dart

```diff
@@ -3,132 +3,43 @@
-              color: MeliColors.black,
-            )));
-  }
-
-  Widget _emptyResult(BuildContext context) {
-    return InfoCard(
-        message: AppLocalizations.of(context)!.paginationListNoResults);
-  }
-
-  Widget _loadMore(BuildContext context, bool isLoading,
-      {required VoidCallback onLoadMore}) {
-    return Column(children: [
-      isLoading
-          ? this._loading()
-          : ElevatedButton(
-              onPressed: onLoadMore,
-              child: Text(AppLocalizations.of(context)!.paginationListLoadMore,
-                  style: const TextStyle(color: MeliColors.black)),
-            )
-    ]);
-  }
-
-  @override
-  Widget build(BuildContext context) {
-    return Query(
-      options: QueryOptions(document: this.paginator.nextPageQuery(null)),
-      builder: (result, {VoidCallback? refetch, FetchMore? fetchMore}) {
-        if (this.paginator.refresh == null) {
-          // Workaround to access `refetch` method from the outside
-          this.paginator.refresh = refetch;
-        }
-
-        if (result.hasException) {
-          return this._error(context, result.exception.toString());
-        }
-
-        if (result.isLoading && result.data == null) {
-          return this._loading();
-        }
-
-        final data =
-            this.paginator.parseJSON(result.data as Map<String, dynamic>);
-
-        FetchMoreOptions opts = FetchMoreOptions(
-          document: this.paginator.nextPageQuery(data.endCursor),
-          updateQuery: (previousResultData, fetchMoreResultData) {
-            return this
-                .paginator
-                .mergeResponses(previousResultData!, fetchMoreResultData!);
-          },
-        );
-
-        if (data.documents.isEmpty) {
-          return this._emptyResult(context);
-        }
-
-        // Deduplicate the returned collection.
-        Set<String> seen = {};
-        List<UsedFor> uniqueUses = data.documents
-            .where((usedFor) => seen.add(usedFor.usedFor))
-            .toList();
-
-        // We want to keep fetching more results (while there are any) until
-        // there are a minimum of 20.
-        if (uniqueUses.length < 20 && data.hasNextPage) {
-          fetchMore!(opts);
-        }
-
-        return Column(
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: <Widget>[
-            this.builder(uniqueUses),
-            if (data.hasNextPage)
-              this._loadMore(context, result.isLoading, onLoadMore: () {
-                fetchMore!(opts);
-              })
-          ],
-        );
-      },
-    );
-  }
-}
 
 class DeduplicatedUsedForTagsList extends StatelessWidget {
   final PaginationBuilder<UsedFor> builder;
   final Paginator<UsedFor> paginator;
+  final minimumPageSize = 10;
 
   const DeduplicatedUsedForTagsList(
       {super.key, required this.builder, required this.paginator});
 
+  bool _fetchMoreOverride(PaginatedCollection<UsedFor> data) {
```

Comment:
We override default behaviour here as we want to remove unavoidable duplicate "uses" from the collection query results.